hl7_transformer/hl7_transformer/application.py

```diff
@@ -74,18 +74,41 @@ def _process_message(
         msh_segment = hl7_msg.msh
         logger.debug(f"Message ID: {msh_segment.msh_10.value}")
 
+        transformations_applied = []
+
         created_datetime = msh_segment.msh_7.value
 
         transformed_datetime = transform_datetime(created_datetime)
         msh_segment.msh_7.value = transformed_datetime
 
+        transformations_applied.append(
+                f"DateTime transformed from {created_datetime} to {transformed_datetime}"
+            )
+
+        if hl7_msg.pid:
+            pid_segment = hl7_msg.pid
+
+            if pid_segment.pid_29.value is not None:
+                original_dod = pid_segment.pid_29.value
```

Comment:
we've encountered nested fields that don't have a `.value` so we had to be defensive and avoid exceptions! Slightly wordier but this is the pattern used elsewhere:
```python
dod_field = getattr(pid_segment, "pid_29", None)
original_dod = (
    getattr(dod_field, "value", dod_field)
    if hasattr(dod_field, "value")
    else dod_field
)
if original_dod is not None:
    ...
```